arrows/klv/applets/compare_klv.cxx

```diff
@@ -0,0 +1,901 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// Definition of compare-klv applet.
+
+#include <arrows/klv/applets/compare_klv.h>
+
+#include <arrows/core/metadata_map_io_csv.h>
+#include <arrows/klv/klv_all.h>
+#include <arrows/klv/klv_key_traits.h>
+#include <arrows/klv/klv_metadata.h>
+
+#include <vital/algo/metadata_map_io.h>
+#include <vital/algo/video_input.h>
+#include <vital/config/config_block_io.h>
+
+#include <filesystem>
+#include <iomanip>
+#include <iostream>
+#include <list>
+
+#include <cfloat>
+
+using namespace kwiver::arrows;
+namespace vital = kwiver::vital;
+using md_map_t = vital::metadata_map::map_metadata_t;
+using frame_t = vital::frame_id_t;
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace klv {
+
+namespace {
+
+// ----------------------------------------------------------------------------
+// Video or JSON file
```

Comment:
At the very least I would suggest this interface and its derived classes be in a separate header and source file. I think we should consider moving the interface to an appropriate place in vital and putting the implementations in `arrows/core`. They are pretty light weight but I can see them being useful to other applications or tools.